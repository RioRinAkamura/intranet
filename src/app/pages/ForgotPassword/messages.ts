/**
 * This file is only need if you want to extract messages into JSON files in locales folder
 * AND if you are also using the object syntax instead of string syntax. \
 * Check the documentation section i18n for details
 */
import { translations } from 'locales/translations';
import { _t } from 'utils/messages';

export const ForgotPageMessages = {
  title: () => _t(translations.ForgotPasswordPage.title, 'Forgot password'),
  description: () =>
    _t(translations.ForgotPasswordPage.description, 'Forgot password'),
  formTitle: () =>
    _t(translations.ForgotPasswordPage.form.title, 'Forgot your password'),
  formDescription: () =>
    _t(
      translations.ForgotPasswordPage.form.description,
      'Enter your registered email below to receive password reset instruction.',
    ),
  emailLabel: () =>
    _t(translations.ForgotPasswordPage.form.email.label, 'Email address'),
  emailPlaceholder: () =>
    _t(translations.ForgotPasswordPage.form.email.placeholder, 'Your email'),
  invalidEmail: () =>
    _t(
      translations.ForgotPasswordPage.form.email.invalid,
      'Your email is invalid',
    ),
  emptyEmail: () =>
    _t(
      translations.ForgotPasswordPage.form.email.empty,
      'Please input your email address',
    ),
  forgotButton: () =>
    _t(translations.ForgotPasswordPage.form.forgotButton, 'Send request'),
  backtButton: () =>
    _t(translations.ForgotPasswordPage.form.backButton, 'Back to login page'),
};
