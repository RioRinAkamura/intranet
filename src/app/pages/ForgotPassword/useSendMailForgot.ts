import { ToastMessageType, useNotify } from 'app/components/ToastNotification';
import React from 'react';
import { useHistory } from 'react-router';
import { api } from 'utils/api';

interface ForgotPasswordPayload {
  email: string;
}

interface ForgotPasswordHook {
  send: (data: ForgotPasswordPayload) => Promise<void>;
  loading: boolean;
  error?: Error | null;
}

export const useSendMailForgot = (): ForgotPasswordHook => {
  const { notify } = useNotify();
  const history = useHistory();
  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState<Error | null>(null);
  const send = async (data: ForgotPasswordPayload): Promise<void> => {
    try {
      setLoading(true);
      await api.auth.resetPassword(data.email);
      notify({
        type: ToastMessageType.Info,
        message: 'Email has been sent',
        duration: 2,
      });
      history.push('/reset-password');
    } catch (e) {
      setError(e);
      notify({
        type: ToastMessageType.Info,
        message: e,
        duration: 2,
      });
    } finally {
      setLoading(false);
    }
  };
  return { send, loading, error };
};
