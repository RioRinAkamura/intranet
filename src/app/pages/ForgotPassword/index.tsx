import { MailTwoTone } from '@ant-design/icons';
import { Col, Row } from 'antd';
import React from 'react';
import { Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components/macro';
import { ForgotForm } from './components/ForgotForm';
import { ForgotPageMessages } from './messages';

export const ForgotPassword: React.FC = () => {
  const { t } = useTranslation();

  return (
    <Wrapper>
      <Helmet>
        <title>{t(ForgotPageMessages.title())}</title>
        <meta
          name="description"
          content={t(ForgotPageMessages.description())}
        />
      </Helmet>
      <ForgotWrapper justify="center" align="middle">
        <Form xxl={6} xl={8} lg={10} xs={21}>
          <Title>
            <h1>{t(ForgotPageMessages.formTitle())}</h1>
          </Title>
          <Description>
            <p>{t(ForgotPageMessages.formDescription())}</p>
            <div>
              <MailTwoTone />
            </div>
          </Description>
          <ForgotForm />
        </Form>
      </ForgotWrapper>
    </Wrapper>
  );
};

export default ForgotPassword;

const Title = styled.div`
  text-align: center;
`;

const ForgotWrapper = styled(Row)`
  position: fixed;
  width: 100%;
  height: 100%;
  overflow: auto;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
`;

const Form = styled(Col)`
  background-color: #fff;
  padding: 2em;
  border-radius: 5px;
  box-shadow: 0px 0px 10px rgb(0 0 0 / 30%);
`;

const Wrapper = styled.div`
  background-color: #ececec;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const Description = styled.div`
  text-align: center;
  color: gray;
  width: 75%;
  margin: 0 auto;
  span {
    font-size: 100px;
    margin: 20px 0;
  }
`;
