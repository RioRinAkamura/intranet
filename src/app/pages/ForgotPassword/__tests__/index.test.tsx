import * as React from 'react';
import { render } from '@testing-library/react';

import ForgotPassword from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<ForgotPassword  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<ForgotPassword />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
