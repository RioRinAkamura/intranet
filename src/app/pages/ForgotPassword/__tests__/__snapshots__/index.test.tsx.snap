// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword  /> should match snapshot 1`] = `
.c6 label {
  font-weight: bold;
}

.c6 div {
  text-align: center;
}

.c5 button {
  width: 100%;
  text-align: center;
  font-weight: 500 !important;
  font-size: 16px !important;
}

.c7 {
  color: gray;
  text-align: center;
  width: 90%;
  margin: 0 auto;
}

.c3 {
  text-align: center;
}

.c1 {
  position: fixed;
  width: 100%;
  height: 100%;
  overflow: auto;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c2 {
  background-color: #fff;
  padding: 2em;
  border-radius: 5px;
  box-shadow: 0px 0px 10px rgb(0 0 0 / 30%);
}

.c0 {
  background-color: #ececec;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c4 {
  text-align: center;
  color: gray;
  width: 75%;
  margin: 0 auto;
}

.c4 span {
  font-size: 100px;
  margin: 20px 0;
}

<div
  class="c0"
>
  <div
    class="ant-row ant-row-center ant-row-middle c1"
  >
    <div
      class="ant-col c2 ant-col-xs-21 ant-col-lg-10 ant-col-xl-8 ant-col-xxl-6"
    >
      <div
        class="c3"
      >
        <h1>
          ForgotPasswordPage.form.title
          Forgot Your Password
        </h1>
      </div>
      <div
        class="c4"
      >
        <p>
          ForgotPasswordPage.form.description
          Enter your registered email below to receive password reset instruction.
        </p>
        <div>
          <span
            aria-label="mail"
            class="anticon anticon-mail"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="mail"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M477.5 536.3L135.9 270.7l-27.5-21.4 27.6 21.5V792h752V270.8L546.2 536.3a55.99 55.99 0 01-68.7 0z"
                fill="#e6f7ff"
              />
              <path
                d="M876.3 198.8l39.3 50.5-27.6 21.5 27.7-21.5-39.3-50.5z"
                fill="#e6f7ff"
              />
              <path
                d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-94.5 72.1L512 482 190.5 232.1h643zm54.5 38.7V792H136V270.8l-27.6-21.5 27.5 21.4 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5h.1l39.3 50.5-27.7 21.5z"
                fill="#1890ff"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="c5"
      >
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-row ant-form-item c6"
          >
            <div
              class="ant-col ant-col-24 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="email"
                title=""
              >
                ForgotPasswordPage.form.email.label
                Email Address
              </label>
            </div>
            <div
              class="ant-col ant-col-24 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                  >
                    <span
                      class="ant-input-prefix"
                    >
                      <span
                        aria-label="user"
                        class="anticon anticon-user"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="user"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                          />
                        </svg>
                      </span>
                    </span>
                    <input
                      class="ant-input ant-input-lg"
                      id="email"
                      placeholder="ForgotPasswordPage.form.email.placeholder,Your email"
                      type="text"
                      value=""
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item c6"
          >
            <div
              class="ant-col ant-col-24 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
                    type="submit"
                  >
                    <span>
                      ForgotPasswordPage.form.forgotButtonSend Request
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <p
            class="c7"
          >
            Did you remember your password?
             
            <div>
              <a
                href="/login"
              >
                ForgotPasswordPage.form.backButton
                Back to login page
              </a>
            </div>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
